src/utils.c

```diff
@@ -642,3 +642,43 @@ DIR *opendir_flags(const char *path, int flags)
 
 	return dirp;
 }
+
+int get_task_personality(pid_t pid, __u32 *personality)
+{
+	__do_close int fd = -EBADF;
+	int ret = -1;
+	char path[STRLITERALLEN("/proc//personality") + INTTYPE_TO_STRLEN(pid_t) + 1];
+	/* seq_printf(m, "%08x\n", task->personality); */
+	char buf[8 + 1];
+
+	snprintf(path, sizeof(path), "/proc/%d/personality", pid);
+	fd = open(path, O_RDONLY | O_CLOEXEC);
+	if (fd < 0)
+		goto err;
+
+	ret = read(fd, buf, sizeof(buf) - 1);
+	if (ret >= 0) {
+		buf[ret] = '\0';
+		*personality = atoi(buf);
```

Comment:
In LXC we have

```c
int lxc_safe_uint(const char *numstr, unsigned int *converted)
{
        char *err = NULL;
        unsigned long int uli;

        while (isspace(*numstr))
                numstr++;

        if (*numstr == '-')
                return -EINVAL;

        errno = 0;
        uli = strtoul(numstr, &err, 0);
        if (errno == ERANGE && uli == ULONG_MAX)
                return -ERANGE;

        if (err == numstr || *err != '\0')
                return -EINVAL;

        if (uli > UINT_MAX)
                return -ERANGE;

        *converted = (unsigned int)uli;
        return 0;
}
```

apparently I've never ported them to LXCFS.

Reply:
ported, fixed ;-)